src/Dictionaries/HashedDictionary.cpp

```diff
@@ -343,6 +347,15 @@ void HashedDictionary::updateData()
     if (!saved_block || saved_block->rows() == 0)
     {
         auto stream = source_ptr->loadUpdatedAll();
+
+        if (preallocate)
```

Comment:
I suppose, there is no need to calculate input size for `HashedDictionary::updateData()` case.
Because we squash all rows into `saved_block` anyway and call `blockToAttributes` only once.
So, the answer is just `saved_block->rows()`.

Reply:
Agree